### PR TITLE
Source Jira: Add caching and more streams

### PR DESCRIPTION
**Pull request by cjwooo**
## What
Reduce the number of duplicate Jira API calls caused by nested streams and add a Boards and Sprints stream.

## How
Use the same VCR caching strategy implemented by the Github Source, and integrate with an older version of the Jira Rest API that surfaces types of records that the latest V2 and V3 APIs do not support.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 1]
```diff
@@ -22,27 +22,56 @@
...
 
+import os
 import urllib.parse as urlparse
 from abc import ABC, abstractmethod
 from typing import Any, Iterable, Mapping, MutableMapping, Optional
 from urllib.parse import parse_qs
 
 import pendulum
 import requests
+import vcr
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:33]: `vcr` is a default library? dont need to add to setup.py?

File: airbyte-integrations/connectors/source-jira/source_jira/schemas/sprints.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      "format": "date-time"
+    },
+    "originBoardId": {
+      "type": "integer"
+    },
+    "goal": {
+      "type": "string"
+    }
+  }
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-jira/source_jira/schemas/sprints.json:36]: EOF, please run ./gradlew format

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Awesome @cjwooo, I added some comments and request @vitaliizazmic to review your contribution too! thanks

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py (hunk 1 above)
Comment (cjwooo) [reply airbyte-integrations/connectors/source-jira/source_jira/streams.py:33]: Oops, I've added it now.

**Review by vitaliizazmic — APPROVED** [review]

**Review by marcosmarxm — APPROVED** [review]
thanks @cjwooo

**Comment by marcosmarxm** [comment]
@cjwooo CI is not passing. Can you send your local tests?
```
2021-10-04 01:11:48 ERROR i.a.c.i.LineGobbler(voidCall):65 - *** -     for message in generator:
2021-10-04 01:11:48 ERROR i.a.c.i.LineGobbler(voidCall):65 - *** -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 108, in read
2021-10-04 01:11:48 ERROR i.a.w.p.a.DefaultAirbyteStreamFactory(internalLog):88 - *** - Encountered an exception while reading stream SourceJira
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 104, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 134, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 206, in _read_full_refresh
    for record in records:
  File "/airbyte/integration_code/source_jira/streams.py", line 405, in read_records
    for property_key in issue_property_keys_stream.read_records(stream_slice=***"key": issue["key"*** **kwargs):
  File "/airbyte/integration_code/source_jira/streams.py", line 386, in read_records
    yield from super().read_records(stream_slice=***"key": issue_key*** **kwargs)
  File "/airbyte/integration_code/source_jira/streams.py", line 100, in read_records
    with self.cache:
  File "/usr/local/lib/python3.7/site-packages/vcr/cassette.py", line 80, in __enter__
    assert self.__finish is None, "Cassette already open."
AssertionError: Cassette already open.
```
https://github.com/airbytehq/airbyte/actions/runs/1301743551

**Comment by cjwooo** [comment]
I test using the same acceptance test files in the repo, but against our own playground Jira instance. I think I've fixed the test errors that were caused by the VCR caching.

**Review by marcosmarxm — APPROVED** [review]
LGTM thanks @cjwooo

**Pull request merged**